src/index.js

```diff
@@ -0,0 +1,28 @@
+const sendCount = require('./utility/sendCount');
+const cron = require('node-cron');
+
+/**
+ * An module to auto send your server counts to botlists.com
+ * @param {*} client Your Discord Bot Client
+ * @param {String} token Your Bot's Token for Website API
+ * @param {Boolean} disableConsole Whether you want console logs or not
+ */
+module.exports = (client, token, disableConsole = false) => {
+    return new Promise((resolve, reject) => {
+        // Invalid Parameter handling
+        if (!client || !"isReady" in client || typeof (client) !== "object") return reject("Invalid Client was provided");
+        if (!token || typeof (token) !== "string" || token.length !== 36) return reject("Invalid Token was provided");
+        if ((typeof (timeout) !== "number" && typeof (timeout) !== "bigint") || timeout < 300000) return reject("Invalid Timeout, Timeout should be in milli seconds and at least greater than 300000");
+        if (typeof (disableConsole) !== "boolean") return reject("Invalid disableConsole type, it should be either false or true");
+
+        // If client is ready i.e. it have its data
+        if (client.isReady()) {
+            cron.schedule('1h', sendCount(client, token, disableConsole));
+        }
+        else {
+            client.once('ready', () => cron.schedule('1h', sendCount(client, token, disableConsole)));
+        }   
+
+        resolve("Auto Server Count Sender is activated");
+    })
```

Comment:
```suggestion
        /* Invalid Parameter handling */
        if (!client || (typeof (client) !== "object" && client !== null && !Array.isArray(client)) ) {
            return reject("Invalid Client was provided");
        }
        if (!token || typeof token !== "string" || token.length !== 36) {
            return reject("Invalid Token was provided");
        }
        if (typeof disableConsole !== "boolean") {
            return reject("Invalid disableConsole type, it should be either false or true");
        }

        /* If client is ready i.e. it have its data */
        if (client.isReady()) {
            cron.schedule('1h', sendCount(client, token, disableConsole));
        } else {
            client.once('ready', () => cron.schedule('1h', sendCount(client, token, disableConsole)));
        }   

        resolve("Auto Server Count Sender is activated");
    });
```

Cleaned up code, and timeout is not even declared. Did you test code prior to sending?

`typeof null` will give you object and `typeof []` as well, so you will need to add more checks like. 

```js
typeof (client) !== "object" && client !== null && !Array.isArray(client)
// checking if type is not a object and not null and not an array. 
```
Edited in suggestion.